### PR TITLE
Engine trash

### LINKED ISSUE #255: Object trash in engine
Currently we have this in `BCFitter`

```
    /** Storage for plot objects with proper clean-up */
    mutable BCAux::BCTrash<TObject> fObjectTrash;
```

But I think we should move it up the hierarchy to `BCEngineMCMC` because it generates similar plot objects that leak. For example in `DrawParameterPlot`

### PR DESCRIPTION
**Pull request by fredRos**
Fixes #255

### CODE REVIEW

File: BAT/BCAux.h [hunk 1]
```diff
@@ -36,6 +36,36 @@ class TH2;
...
+ * transferred in copies, only during a swap. */
+template <typename T>
+class BCTrash
+{
+public:
+    BCTrash() {}
+    BCTrash(const BCTrash<T>&) {}
+    BCTrash<T>& operator=(const BCTrash<T>&)
+    {
+        return *this;
```
Comment (greenwald) [inline BAT/BCAux.h]: What is the meaning of this?

File: src/BCAux.cxx [hunk 2]
```diff
@@ -354,7 +354,7 @@ void BCAux::SetKnowledgeUpdateDrawingStyle(BCH2D& prior, BCH2D& posterior, BCAux
 
 
 // ---------------------------------------------------------
-void BCAux::DrawKnowledgeUpdate(BCHistogramBase& prior, BCHistogramBase& posterior, bool draw_prior_first)
+void BCAux::DrawKnowledgeUpdate(BCHistogramBase& prior, BCHistogramBase& posterior, bool draw_prior_first, BCTrash<TObject>& trash)
```
Comment (greenwald) [inline src/BCAux.cxx]: If we want users to also use this code on their own, I think we need to have an example in the `examples` directory that shows the creation and use of a `BCTrash` object.

File: BAT/BCAux.h (hunk 1 above)
Comment (fredRos) [reply BAT/BCAux.h]: As it says in the docstring, copy and assignment don't do anything. But they are implemented following the general interface so `BCTrash` plays well as a member of another class with compiler-generated copy ctor etc.

File: src/BCAux.cxx (hunk 2 above)
Comment (fredRos) [reply src/BCAux.cxx]: You are right but I don't want users to use this on their own.

**Pull request merged**